tests/onnxruntime/test_modeling_ort.py

```diff
@@ -481,3 +504,107 @@ def test_pipeline(self, *args, **kwargs):
+    }
+
+    @parameterized.expand(SUPPORTED_ARCHITECTURES_WITH_MODEL_ID.items())
+    def test_supported_transformers_architectures(self, *args, **kwargs):
+        model_arch, model_id = args
+        model = ORTModelForSeq2SeqLM.from_pretrained(model_id, from_transformers=True)
+        self.assertIsInstance(model.model, ORTModelForConditionalGeneration)
+        self.assertIsInstance(model.config, PretrainedConfig)
+
+    def test_load_vanilla_transformers_which_is_not_supported(self):
+        with self.assertRaises(Exception) as context:
+            model = ORTModelForSeq2SeqLM.from_pretrained("t5-small", from_transformers=False)
+        self.assertTrue("Unrecognized configuration class", context.exception)
+
+    @parameterized.expand(SUPPORTED_ARCHITECTURES_WITH_MODEL_ID.items())
+    def test_model_call(self, *args, **kwargs):
+        model_arch, model_id = args
+        model = ORTModelForSeq2SeqLM.from_pretrained(model_id, from_transformers=True)
+        tokenizer = AutoTokenizer.from_pretrained(model_id)
+        tokens = tokenizer("This is a sample output", return_tensors="pt")
+        decoder_start_token_id = model.config.decoder_start_token_id
+        decoder_inputs = {"decoder_input_ids": torch.ones((1, 1), dtype=torch.long) * decoder_start_token_id}
+        outputs = model(**tokens, **decoder_inputs)
+        self.assertTrue("logits" in outputs)
+        self.assertTrue(isinstance(outputs.logits, torch.Tensor))
+
+    @parameterized.expand(SUPPORTED_ARCHITECTURES_WITH_MODEL_ID.items())
+    def test_generate_utils(self, *args, **kwargs):
+        model_arch, model_id = args
+        model = ORTModelForSeq2SeqLM.from_pretrained(model_id, from_transformers=True)
+        tokenizer = AutoTokenizer.from_pretrained(model_id)
+        text = "This is a sample output"
+        tokens = tokenizer(text, return_tensors="pt")
+        outputs = model.generate(**tokens)
+        res = tokenizer.batch_decode(outputs, skip_special_tokens=True)
+        self.assertTrue(isinstance(res[0], str))
+
+    @parameterized.expand(SUPPORTED_ARCHITECTURES_WITH_MODEL_ID.items())
+    def test_generate_utils_with_input_ids(self, *args, **kwargs):
+        model_arch, model_id = args
+        model = ORTModelForSeq2SeqLM.from_pretrained(model_id, from_transformers=True)
+        tokenizer = AutoTokenizer.from_pretrained(model_id)
+        text = "This is a sample output"
+        tokens = tokenizer(text, return_tensors="pt")
+        outputs = model.generate(input_ids=tokens["input_ids"])
+        res = tokenizer.batch_decode(outputs, skip_special_tokens=True)
+        self.assertTrue(isinstance(res[0], str))
+
+    @parameterized.expand(SUPPORTED_ARCHITECTURES_WITH_MODEL_ID.items())
+    def test_compare_to_transformers(self, *args, **kwargs):
+        model_arch, model_id = args
+        onnx_model = ORTModelForSeq2SeqLM.from_pretrained(model_id, from_transformers=True)
+        trfs_model = AutoModelForSeq2SeqLM.from_pretrained(model_id)
```

Comment:
nit: I think it would be nice to be a bit more explicit in the variable naming here

```suggestion
        transformers_model = AutoModelForSeq2SeqLM.from_pretrained(model_id)
```